### PR TITLE
[FIX] survey: correctly consider all non-answered questions as skipped

### PR DESCRIPTION
**Pull request by awa-odoo**
Currently, if the survey participant does not answer questions of type:
- simple_choice
- multiple_choice
- matrix

The system does not register any survey.user_input_line with the "skipped"
attribute set to True. Meaning that this question's answer will not appear in
the survey statistics as skipped (in fact it will not appear at all).

This commit makes sure we correctly save a user_input_line set as skipped=True
when not answering to those types of questions.

A unit test has been added to make sure that we consider every non-answered
question type as properly skipped within the survey statistics.

Task-2622869

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by std-odoo — COMMENTED** [review]
Just one small remark, otherwise LGTM :)

File: addons/survey/tests/test_survey.py [hunk 1]
```diff
@@ -113,3 +103,47 @@ def test_answer_validation_char_box_length(self):
...
+    # TOOLS / UTILITY
+    # ------------------------------------------------------------
+
+    def _assert_skipped_question(self, question, survey_user):
+        statistics = question._prepare_statistics(survey_user.user_input_line_ids)
+        question_data = next(
+            question_data
+            for question_data in statistics
+            if question_data.get('question') == question
+        )
```
[the remark this answers is not in the document]
Comment (awa-odoo) [reply addons/survey/tests/test_survey.py]: It's a bit more explicit to have the test fail at "assertTrue(False)" than for a "StopIteration" so yes, good idea :)

File: addons/survey/tests/test_survey.py [hunk 2]
```diff
@@ -113,3 +103,48 @@ def test_answer_validation_char_box_length(self):
             question.validate_question('valid'),
             {}
         )
+
+    def test_skipped_values(self):
```
Comment (tde-banana-odoo) [inline addons/survey/tests/test_survey.py:107]: Would add survey_manager user decorator, never perdu.

**Review by tde-banana-odoo — COMMENTED** [review]
Riviou ! LGTM, some small remarks :)

File: addons/survey/models/survey_question.py [hunk 3]
```diff
@@ -392,7 +392,7 @@ def _prepare_statistics(self, user_input_lines):
             all_lines = user_input_lines.filtered(lambda line: line.question_id == question)
             if question.question_type in ['simple_choice', 'multiple_choice', 'matrix']:
                 answer_lines = all_lines.filtered(
-                    lambda line: line.answer_type == 'suggestion' or (
+                    lambda line: not line.answer_type or line.answer_type == 'suggestion' or (
```
[the remark this answers is not in the document]
Comment (awa-odoo) [reply addons/survey/models/survey_question.py:395]: In the method "_get_line_answer_values", we force the answer_type to None for skipped answers.

Meaning here, we want to check answer_lines that are either "None", or "suggestion", or "char_box + comment".
So yes, we do want those records with no answer_type to be able to correctly do `skipped_lines = answer_lines.filtered(lambda line: line.skipped)` here below.

Maybe I did not understand your question?

[the remark this answers is not in the document]
Comment (tde-banana-odoo) [reply addons/survey/models/survey_question.py:395]: Idea was to explicitely say: ``... (line.skipped and not line.answer_type) or line.answer_type == 'suggestion' or ... ``

[the remark this answers is not in the document]
Comment (awa-odoo) [reply addons/survey/models/survey_question.py]: Oh ok I see what you mean.
It should not change anything but yes, better safe than sorry!

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-ff

**Comment by tde-banana-odoo** [comment]
@awa-odoo seems upgrade is down, probably an issue with branches ?

**Comment by awa-odoo** [comment]
@tde-banana-odoo 

Considering the error says
> odoo.modules.graph:68 module sale_ebay_account_deletion: not installable, skipped

I hope so :smile: 
I rebased, we will see what happens.

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-ff

**Pull request closed without merging**